app/models/grade_entry_form.rb

```diff
@@ -44,44 +44,56 @@ def out_of_total
 
   # Determine the total mark for a particular student, as a percentage
   def calculate_total_percent(grade_entry_student)
-    unless grade_entry_student.nil?
-      total = grade_entry_student.total_grade
+    unless grade_entry_student.nil? || !@ges_total_grade.nil?
+      @ges_total_grade = grade_entry_student.total_grade
     end
 
     percent = BLANK_MARK
     out_of = self.out_of_total
 
-    # Check for NA mark or division by 0
-    unless total.nil? || out_of == 0
-      percent = (total / out_of) * 100
+    # Check for NA mark f or division by 0
+    unless @ges_total_grade.nil? || out_of == 0
+        percent = (@ges_total_grade / out_of) * 100
     end
+
     percent
   end
 
   # An array of all grade_entry_students' released percentage total grades that are not nil
   def released_percentage_grades_array
     grades = Array.new()
-    grade_entry_students = self.grade_entry_students
-                             .where(released_to_student: true)
-    grade_entry_students.each do |grade_entry_student|
-      if !grade_entry_student.total_grade.nil?
+    ges = GradeEntryStudent.includes(:grades)
+                           .where(released_to_student: true,
+                                  grade_entry_form: self)
+
+    ges.each do |grade_entry_student|
+      @ges_total_grade = grade_entry_student.total_grade
+      unless @ges_total_grade.nil?
         grades.push(calculate_total_percent(grade_entry_student))
       end
     end
 
-    return grades
+    @released_grades_array = grades
+
+    grades
   end
 
   # An array of all grade_entry_students' percentage total grades that are not nil
   def percentage_grades_array
-    grades = Array.new
-    grade_entry_students.each do |grade_entry_student|
-      if !grade_entry_student.total_grade.nil? && out_of_total > 0
+    grades = Array.new()
+    ges = GradeEntryStudent.includes(:grades)
+                           .where(grade_entry_form: self)
+
+    ges.each do |grade_entry_student|
+      @ges_total_grade = grade_entry_student.total_grade
```

Comment:
On the other hand, it is *not* appropriate to make `ges_total_grade` an instance variable. It should be local to this method only. (Similar problem occurs above.)